dxr/plugins/xpidl/visitor.py

```diff
@@ -0,0 +1,252 @@
+from cStringIO import StringIO
+from itertools import ifilter
+from os.path import relpath, join, basename, dirname, exists
+
+from flask import url_for
+from xpidl.header import (idl_basename, header, include, jsvalue_include,
+                          infallible_includes, header_end, forward_decl,
+                          write_interface, printComments)
+from xpidl.xpidl import Attribute
+
+from dxr.indexers import Ref, Extent, Position
+from dxr.plugins.xpidl.filters import PLUGIN_NAME
+from dxr.utils import search_url
+
+
+def start_pos(name, location):
+    """Return the last byte position on which we can find name in the
+    location's line, resolving if necessary."""
+
+    location.resolve()
+    return location._line.rfind(name) - location._colno + location._lexpos
+
+
+def header_line_numbers(idl, filename):
+    """Map each production to its line number in the header."""
+    # Basically I replace all fd.write in print_header with line increments
```

Comment:
Can you make more clear that you ripped this off from elsewhere, where that is, and that it's not your fault? ;-)